src/Middleware/ConcurrencyLimiter/src/QueuePolicies/ResettableBooleanCompletionSource.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Sources;
+
+namespace Microsoft.AspNetCore.ConcurrencyLimiter
+{
+    internal class ResettableBooleanCompletionSource : IValueTaskSource<bool>
```

Comment:
nit: Add a comment with a general description for this class.

Reply:
```
    /// Custom awaiter to allow the StackPolicy to reduce allocations.
    /// When this completion source has its result checked, it resets itself and returns itself to the cache of its parent StackPolicy.
    /// Then when the StackPolicy needs a new completion source, it tries to get one from its cache, otherwise it allocates.
```

This seem good?

Reply:
Seems good to me